src/Domain/Creators/Rectangle.hpp

```diff
@@ -75,21 +77,50 @@ class Rectangle : public DomainCreator<2> {
     static constexpr Options::String help = {
         "The time dependence of the moving mesh domain."};
   };
+  template <typename BoundaryConditionsBase>
+  struct BoundaryCondition {
```

Comment:
Do I understand correctly that you're adding basic boundary condition support to impose the same conditions on all sides, and then we can generalize this later to treat each side differently? (I think that's a good strategy 👍)

Reply:
Yep, that's right! There are just way too many possible combinations, especially for domain creators like 3d aligned lattice, that I think it's better to just wait until someone needs a specific config to implement.